Assets/Minigames/Sorting/Scripts/Sound/MusicController.cs

```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class MusicController : MonoBehaviour
+{
+    public GameObject audioManager;
+    public Sprite offSprite;
+    public Sprite onSprite;
+
+    private Image buttonImage;
+    private bool isPlaying = true;
+
+    public void ToggleMusic()
+    {
+        buttonImage = GetComponent<Image>();
```

Comment:
Don't use GetComponent if you can assign it in the inspector. And definitely don't GetComponent every time the function is called. If you need to use it then do that once in Awake or Start and cache the reference in a variable (buttonImage which you're already using)

In this case, assign the component to the variable in the Inspector. If you want to keep the variable private then add `[SerializeField]` to it
```cs
[SerializeField]
private Image buttonImage;
```